### PR TITLE
BAQE-1494 - Migrate jBPM Performance Tests

### PR DESCRIPTION
**Pull request by MichalCervenansky**
This module uses the JMH framework as a basis for the jbpm engine scenarios kept in org.jbpm.test.performance.scenario.

### CODE REVIEW

File: jbpm-benchmarks/jbpm-performance-tests-jmh/src/main/java/org/jbpm/test/performance/jbpm/JBPMTestConfig.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    public boolean isHumanTaskEager() {
+        return humanTaskEager;
+    }
+
+    protected void addTag(String tag) {
+        if (!tags.contains(tag)) {
+            tags.add(tag);
+        }
+    }
```
Comment (MarianMacik) [inline jbpm-benchmarks/jbpm-performance-tests-jmh/src/main/java/org/jbpm/test/performance/jbpm/JBPMTestConfig.java:92]: I still think we don't need tags as they are not used anymore.

File: jbpm-benchmarks/jbpm-performance-tests/src/main/resources/processes/MortgageApplication.bpmn2 [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <bpmn2:definitions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.omg.org/bpmn20" xmlns:bpmn2="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:bpsim="http://www.bpsim.org/schemas/1.0" xmlns:color="http://www.omg.org/spec/BPMN/non-normative/color" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:drools="http://www.jboss.org/drools" id="_6M964MtNEeWFGc2OmJDzFA" xsi:schemaLocation="http://www.omg.org/spec/BPMN/20100524/MODEL BPMN20.xsd http://www.jboss.org/drools drools.xsd http://www.bpsim.org/schemas/1.0 bpsim.xsd" exporter="jBPM Designer" exporterVersion="6.2.0" expressionLanguage="http://www.mvel.org/2.0" targetNamespace="http://www.omg.org/bpmn20" typeLanguage="http://www.java.com/javaTypes">
   <bpmn2:itemDefinition id="_ssnItem" structureRef="Integer"/>
   <bpmn2:itemDefinition id="_downPaymentItem" structureRef="Integer"/>
-  <bpmn2:itemDefinition id="_applicationItem" structureRef="com.redhat.bpms.examples.mortgage.Application"/>
```
Comment (MarianMacik) [inline jbpm-benchmarks/jbpm-performance-tests/src/main/resources/processes/MortgageApplication.bpmn2:5]: This change is not needed, this is in the other module.

File: jbpm-benchmarks/jbpm-performance-tests/src/main/java/org/jbpm/test/performance/scenario/PrepareEngine.java [hunk 3]
```diff
@@ -9,14 +9,15 @@
...
     public static List<Long> createNewTasks(boolean start, int count, TaskService taskService, String runtimeManagerIdentifier) {
         String str = "(with (new Task()) { priority = 55, taskData = (with( new TaskData()) { deploymentId = '" + runtimeManagerIdentifier + "' } ), ";
         str += "peopleAssignments = (with ( new PeopleAssignments() ) { potentialOwners = [new User('" + UserStorage.PerfUser.getUserId()
                 + "')], businessAdministrators = [ new User('Administrator') ], }),";
         str += "names = [ new I18NText( 'en-UK', 'perf-sample-task')] })";
-        List<Long> taskIds = new ArrayList<Long>();
+        List<Long> taskIds = new ArrayList<>();
         while (count > 0) {
             Task task = TaskFactory.evalTask(new StringReader(str));
             long taskId = taskService.addTask(task, null);
```
Comment (MarianMacik) [inline jbpm-benchmarks/jbpm-performance-tests/src/main/java/org/jbpm/test/performance/scenario/PrepareEngine.java:23]: This file doesn't need to be changed at all.

File: jbpm-benchmarks/jbpm-performance-tests/src/main/resources/processes/MortgageApplication.bpmn2 (hunk 2 above)
Comment (MarianMacik) [reply jbpm-benchmarks/jbpm-performance-tests/src/main/resources/processes/MortgageApplication.bpmn2:5]: Same here, still affected. Please remove these changes from the PR.

File: jbpm-benchmarks/jbpm-performance-tests/src/main/java/org/jbpm/test/performance/scenario/PrepareEngine.java (hunk 3 above)
Comment (MarianMacik) [reply jbpm-benchmarks/jbpm-performance-tests/src/main/java/org/jbpm/test/performance/scenario/PrepareEngine.java:23]: This file is still shown as changed but as it is in the other module I wouldn't change it in this PR.

File: jbpm-benchmarks/jbpm-performance-tests/pom.xml [hunk 4]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.jbpm</groupId>
     <artifactId>jbpm-benchmarks</artifactId>
-    <version>7.41.0-SNAPSHOT</version>
+    <version>7.37.0-SNAPSHOT</version>
```
Comment (MarianMacik) [inline jbpm-benchmarks/jbpm-performance-tests/pom.xml:8]: Not needed.

File: jbpm-benchmarks/jbpm-performance-tests/src/main/resources/processes/MortgageApplication.bpmn2 (hunk 2 above)
Comment (MarianMacik) [reply jbpm-benchmarks/jbpm-performance-tests/src/main/resources/processes/MortgageApplication.bpmn2:5]: This file is still here :D

File: jbpm-benchmarks/jbpm-performance-tests-jmh/src/main/java/org/jbpm/test/performance/scenario/PrepareEngine.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package org.jbpm.test.performance.scenario;
+
+import org.jbpm.test.performance.jbpm.constant.UserStorage;
+import org.jbpm.services.task.impl.factories.TaskFactory;
+import org.kie.api.task.TaskService;
+import org.kie.api.task.model.Task;
+
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
```
Comment (MarianMacik) [inline jbpm-benchmarks/jbpm-performance-tests-jmh/src/main/java/org/jbpm/test/performance/scenario/PrepareEngine.java:10]: The order of imports should be like before - first java imports then other packages.

File: jbpm-benchmarks/jbpm-performance-tests/src/main/resources/processes/MortgageApplication.bpmn2 (hunk 2 above)
Comment (MichalCervenansky) [reply jbpm-benchmarks/jbpm-performance-tests/src/main/resources/processes/MortgageApplication.bpmn2]: Hopefully it should be okey now.

**Pull request merged**